src/Xamarin.Android.Build.Tasks/Utilities/ResourceDesignerImportGenerator.cs

```diff
@@ -54,6 +66,11 @@ void CreateImportFor (bool isNestedSrc, IEnumerable<TypeDefinition> types, CodeM
 				foreach (var field in type.Fields) {
 					var dstField = new CodeFieldReferenceExpression (dstClassRef, field.Name);
 					var srcField = new CodeFieldReferenceExpression (srcClassRef, field.Name);
+					var fieldName = $"{primary_name}.{type.Name}.{field.Name}";
+					if (!resourceFields.Contains (fieldName)) {
+						Log.LogWarning (null, "XA0106", null, null, lineNumber: 0, columnNumber: 0, endLineNumber: 0, endColumnNumber: 0, message: $"Skipping {fieldName}. Check your Nuget Package versions are compatible.");
```

Comment:
All parameters should be named here, because nobody's going to remember what any of them are.

For readability, each parameter should go on a separate line. See e.g. [`Generator`](https://github.com/xamarin/xamarin-android/blob/28350a3/src/Xamarin.Android.Build.Tasks/Generator/Generator.cs#L23-L34).